js/widgets/modewidget.js

```diff
@@ -9,152 +9,145 @@
 // License along with this library; if not, write to the Free Software
 // Foundation, 51 Franklin Street, Suite 500 Boston, MA 02110-1335 USA
 
-function ModeWidget() {
-    const ROTATESPEED = 125;
-    const BUTTONDIVWIDTH = 535;
-    const BUTTONSIZE = 53;
-    const ICONSIZE = 32;
+class ModeWidget {
+
+    get CONSTANTS() {
+        return {
+            ICONSIZE: 32,
+            BUTTONSIZE: 53,
+            ROTATESPEED: 125,
+            BUTTONDIVWIDTH: 535
+        }
+    }
 
-    this.init = function(logo, modeBlock) {
+    constructor() {
         this._logo = logo;
-        this._modeBlock = modeBlock;
+        this._modeBlock = logo._modeBlock;
```

Comment:
directly use `logo._modeBlock` wherever required? Plus remove the leading underscore inside logo. I know the conventions here are messed up but by convention leading underscore variables are private ones. Let's try clean this up as we go. Would make it easier for everyone during porting to Music Blocks 2.